### PR TITLE
(RE-9889) `legacy` wasn't maintaining legacy behavior

### PR DESCRIPTION
**Pull request by underscorgan**
In an effort to maintain functionality of current nightlies pipelines, a
`legacy` option was added to Pkg::Paths.repo_path(). However, this did
not maintain behavior from packaging#master
Pkg::Util::Platform.repo_path(), which is causing existing nightly
pipelines to only ship repo configs, not actual repos/packages.

### CODE REVIEW

File: lib/packaging/paths.rb [hunk 1]
```diff
@@ -178,7 +178,7 @@ def artifacts_path(platform_tag, path_prefix = 'artifacts')
 
   def repo_path(platform_tag, options = { :legacy => false })
     repo_target = repo_name
-    repo_target = Pkg::Config.repo_name if options[:legacy]
+    repo_target = '**' if options[:legacy]
```
Comment (shrug) [inline lib/packaging/paths.rb]: https://media.giphy.com/media/Fwn1auHR2LUGI/giphy.gif

File: lib/packaging/paths.rb [hunk 2]
```diff
@@ -178,7 +178,12 @@ def artifacts_path(platform_tag, path_prefix = 'artifacts')
 
   def repo_path(platform_tag, options = { :legacy => false })
     repo_target = repo_name
-    repo_target = Pkg::Config.repo_name if options[:legacy]
+    # in legacy packaging methods, there was no consistent way to determine the
+    # repo name. There were separate variables for apt_repo_name and
+    # yum_repo_name. At times, either or both of these were unset, and they had
+    # different defaults. So, for legacy automation we need to just use a splat
+    # and globbing to find our packages.
+    repo_target = '**' if options[:legacy]
```
Comment (shrug) [inline lib/packaging/paths.rb]: hungry_hippos

Comment (e-gris) [reply lib/packaging/paths.rb]: https://gph.is/1KP1eHi

Comment (sarameisburger) [reply lib/packaging/paths.rb]: I really appreciate the descriptive comment!

**Pull request merged**